export/export.go

```diff
@@ -0,0 +1,432 @@
+package export
+
+//go:generate mockgen -destination=./mock_$GOPACKAGE/pod.go . PodService
+//go:generate mockgen -destination=./mock_$GOPACKAGE/node.go . NodeService
+//go:generate mockgen -destination=./mock_$GOPACKAGE/pv.go . PersistentVolumeService
+//go:generate mockgen -destination=./mock_$GOPACKAGE/pvc.go . PersistentVolumeClaimService
+//go:generate mockgen -destination=./mock_$GOPACKAGE/storageClassc.go . StorageClassService
+//go:generate mockgen -destination=./mock_$GOPACKAGE/scheduler.go . SchedulerService
+//go:generate mockgen -destination=./mock_$GOPACKAGE/priorityclass.go . PriorityClassService
+
+import (
+	"context"
+
+	"golang.org/x/xerrors"
+	corev1 "k8s.io/api/core/v1"
+	schedulingv1 "k8s.io/api/scheduling/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	v1 "k8s.io/client-go/applyconfigurations/core/v1"
+	schedulingcfgv1 "k8s.io/client-go/applyconfigurations/scheduling/v1"
+	confstoragev1 "k8s.io/client-go/applyconfigurations/storage/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+	v1beta2config "k8s.io/kube-scheduler/config/v1beta2"
+
+	"github.com/kubernetes-sigs/kube-scheduler-simulator/util"
+)
+
+type Service struct {
+	client               clientset.Interface
+	podService           PodService
+	nodeService          NodeService
+	pvService            PersistentVolumeService
+	pvcService           PersistentVolumeClaimService
+	storageClassService  StorageClassService
+	priorityclassService PriorityClassService
+	schedulerService     SchedulerService
+}
+
+// ResourcesForExport denotes all resources and scheduler configuration for export.
+type ResourcesForExport struct {
+	Pods            []corev1.Pod                              `json:"pods"`
+	Nodes           []corev1.Node                             `json:"nodes"`
+	Pvs             []corev1.PersistentVolume                 `json:"pvs"`
+	Pvcs            []corev1.PersistentVolumeClaim            `json:"pvcs"`
+	StorageClasses  []storagev1.StorageClass                  `json:"storageClasses"`
+	PriorityClasses []schedulingv1.PriorityClass              `json:"priorityClasses"`
+	SchedulerConfig *v1beta2config.KubeSchedulerConfiguration `json:"schedulerConfig"`
+}
+
+// ResourcesForImport denotes all resources and scheduler configuration for import.
+type ResourcesForImport struct {
+	Pods            []v1.PodApplyConfiguration                        `json:"pods"`
+	Nodes           []v1.NodeApplyConfiguration                       `json:"nodes"`
+	Pvs             []v1.PersistentVolumeApplyConfiguration           `json:"pvs"`
+	Pvcs            []v1.PersistentVolumeClaimApplyConfiguration      `json:"pvcs"`
+	StorageClasses  []confstoragev1.StorageClassApplyConfiguration    `json:"storageClasses"`
+	PriorityClasses []schedulingcfgv1.PriorityClassApplyConfiguration `json:"priorityClasses"`
+	SchedulerConfig *v1beta2config.KubeSchedulerConfiguration         `json:"schedulerConfig"`
+}
+type ErrGroupWithSemaphore util.ErrGroupWithSemaphore
```

Comment:
I think this type alias is not needed. 
Let's remove this and use `util.ErrGroupWithSemaphore` directly.  
`listHoge` and `applyHoge` currently use `*ErrGroupWithSemaphore` as method receiver; change these functions to receive `*util.ErrGroupWithSemaphore` in args.